spec/client/Rollup-Integration-guide.md

```diff
@@ -0,0 +1,201 @@
+
+This guide is not intended to be a formal specification or Interchain Standard. As the architectures for rollups and their underlying data availability and settlement layers differ vastly: from ZK rollups to optimistic rollups with separate data availability and settlement layers to sovereign rollups; it is impossible to write a fully specified client to encompass all these cases. Thus this guide is intended to highlight the client functions that are most affected by rollup specific features and explain what must be done in each one to take into account the unique properties of rollups. Rollup light client developers should use this document as a starting point when desigining their light clients to ensure they are taking into account rollup-specific logic in the appropriate places.
+
+### `verifyClientMessage`
+
+In order to verify a new header for the rollup, the rollup client must also be able to verify the header's (and associated block's) inclusion in the DA layer. Thus, the rollup client's update logic **must** have the ability to execute verification of the DA client. After verifying the rollups own consensus mechanism (which itself may be nonexistent for some rollup architectures), it verifies the header and blockdata in the data availability layer. Simply proving inclusion is not enough however, we must ensure that the data we are proving is valid; i.e. the data is not simply included but is included in the way that is expected by the rollup architecure. In the example below, we check that the blockdata hashes to the `txHash` in the header.
+
+ZK rollups can verify correctness of the header upon submission since the rollup client can embed a proving circuit that can verify a ZK proof from the relayer that the submitted header is correct. Optimistic rollups on the other hand cannot immediately trust a header upon submission, as the header may later be proved fraudulent. Thus, the header can be stored but must wait for the fraud period to elapse without any successful challenges to the correctness of the header before it is finalized and used for proof verification.
+
+```typescript
+function verifyClientMessage(clientMessage: ClientMessage) {
+    clientState = getClient(clientMessage.clientId)
+    // note: unmarshalling logic omitted
+    // verify the header against the rollups own consensus mechanism if it exists
+    // e.g. verify sequencer signature
+    verifySignatures(clientMessage.header, clientSequencers)
```

Comment:
Do we need to do any verifications is `clientMessage` is actually of `Misbehaviour` type?

Reply:
Fixed